src/onboarding/steps.test.ts

```diff
@@ -0,0 +1,254 @@
+import { BIOMETRY_TYPE } from 'react-native-keychain'
+import { initializeAccount } from 'src/account/actions'
+import { setHasSeenVerificationNux } from 'src/identity/actions'
+import {
+  navigate,
+  navigateClearingStack,
+  navigateHome,
+  popToScreen,
+} from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { getOnboardingStepValues, goToNextOnboardingScreen } from 'src/onboarding/steps'
+import { store } from 'src/redux/store'
+import { mockOnboardingProps } from 'test/values'
+import { mocked } from 'ts-jest/utils'
+
+jest.mock('src/redux/store', () => ({ store: { dispatch: jest.fn() } }))
+
+const mockStore = mocked(store)
+
+describe('onboarding steps', () => {
+  const newUserFlowWithEverythingEnabled = {
+    onboardingProps: {
+      ...mockOnboardingProps,
+      skipVerification: false,
+      supportedBiometryType: BIOMETRY_TYPE.FACE_ID,
+      recoveringFromStoreWipe: false,
+    },
+    screens: [
+      Screens.NameAndPicture,
+      Screens.PincodeSet,
+      Screens.EnableBiometry,
+      Screens.VerificationStartScreen,
+    ],
+    name: 'newUserFlowWithEverythingEnabled',
+  }
+
+  const newUserFlowWithEverythingDisabled = {
+    onboardingProps: {
+      ...mockOnboardingProps,
+      skipVerification: true,
+      supportedBiometryType: null,
+      recoveringFromStoreWipe: false,
+    },
+    screens: [Screens.NameAndPicture, Screens.PincodeSet],
+    name: 'newUserFlowWithEverythingDisabled',
+  }
+
+  const importWalletFlowEverythingEnabled = {
+    onboardingProps: {
+      ...mockOnboardingProps,
+      recoveringFromStoreWipe: false,
+      choseToRestoreAccount: true,
+      skipVerification: false,
+      supportedBiometryType: BIOMETRY_TYPE.FACE_ID,
+    },
+    screens: [
+      Screens.NameAndPicture,
+      Screens.PincodeSet,
+      Screens.EnableBiometry,
+      Screens.ImportWallet,
+      Screens.VerificationStartScreen,
+    ],
+    name: 'importWalletFlowEverythingEnabled',
+  }
+
+  it.each([
+    newUserFlowWithEverythingEnabled,
+    newUserFlowWithEverythingDisabled,
+    importWalletFlowEverythingEnabled,
+  ])(
+    'goToNextOnboardingScreen and getOnboardingStepValues work as expected',
+    ({ onboardingProps, screens, name }) => {
+      const expectedTotalSteps = screens.length
+      screens.forEach((screen, index) => {
+        const { totalSteps, step } = getOnboardingStepValues(screen, onboardingProps)
+        // Checking that the step number is correct
+        try {
+          expect(step).toEqual(index + 1)
+          expect(totalSteps).toEqual(expectedTotalSteps)
+        } catch (error) {
+          throw new Error(
```

Comment:
nit: jest already emits a similar error message

Reply:
maybe you were worried about figuring out which test failed. but you should be able to inject parameters from the test cases into the test description. (that being said, I tried adding "$name" to the end of the test description and it didn't work-- not sure why. maybe the jest version?)

https://jestjs.io/docs/26.x/api#testeachtablename-fn-timeout